rust/noosphere-gateway/src/middleware.rs

```diff
@@ -0,0 +1,244 @@
+use anyhow::Result;
```

Comment:
Please make this a folder with an `authority.rs` module inside. I've realized recently that naming the file after what it contains is key to quickly navigating this code base, and I probably won't remember to look for "middleware" when I'm trying to find the "gateway authority."